### PR TITLE
Update RAI backends and installation

### PR DESCRIPTION
**Pull request by Spartee**
This PR includes the following
 - Update RedisAI backend to 1.2.3 which enables
     - PyTorch 1.7.1
     - TensorFlow 2.4.0
     - ONNX 1.6
   
The PyTorch backend is no longer listed in the dev requirements as it is used for the backend for RedisAI within SmartSim. The ``smart`` cli will now download (through pip) the PyTorch backend and use the shared libraries to build the ML backend for RedisAI. This makes
   - the install faster
   - dockerfiles smaller
   - enables systems with older versions of GlibC to get around using the pre-builts

Also included are
 - Tests for the TensorFlow/Keras backend https://github.com/craylabs/smartredis/issues/56
 - Tests for the Torch backend 
 - Multiple tests for the ONNX backend with sklearn https://github.com/craylabs/smartredis/issues/57
 - utilizes for serializing and freezing a TensorFlow graph/model.

Minor additions:
 - cli can now emit the smartsim installation site ``smart --site``
 - cli can not emit the redis-cli location ``smart --dbcli``, a slightly easier and more maintainable approach then the one listed in #64

### CODE REVIEW

File: smart [hunk 1]
```diff
@@ -135,6 +321,11 @@ def make_description():
     other += "build and install other non-python dependencies\n"
     other += "needed by SmartSim."
```
Comment (al-rigazzi) [inline smart:323]: -> `"If backends are not built, users will still be able\n"`

**Review by al-rigazzi — CHANGES_REQUESTED** [review]
Only minor comments except for the the versioning which I'm not sure about. Otherwise looks good to merge -- really exciting progress

File: smartsim/tf/__init__.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    import tensorflow as tf
+
+    # in python >= 3.9 Tensorflow only supports >= tf 2.5
+    if sys.version_info.minor >= 9:
+        logger.warning(
+            f"TensorFlow >= 2.5 is not offcially supported, SmartSim uses TensorFlow {TF_VERSION}"
+        )
+
+    tf_version = tf.__version__.split(".")
+    assert tf_version[0] == "2" and tf_version[1] >= "4"
```
[the remark this answers is not in the document]
Comment (Spartee) [reply smartsim/tf/__init__.py:19]: Good catch

**Review by al-rigazzi — APPROVED** [review]
L-G-T-M!

**Pull request merged**